client.js

```diff
@@ -111,7 +112,11 @@ module.exports = function(opts) {
 
         // regsiter the pending callback
         pendingCallbacks[id] = cb;
-        window.postMessage({
+
+        // Determine the target window for the message
+        // (if we are a frame, and operating in a frame is allowed, we need to pass it up)
+        var target = (window.parent ? window.parent : window);
+        target['postMessage']({
```

Comment:
nit: Any reason for the `target['postMessage']` syntax over `target.postMessage` ?